### PR TITLE
Infer logical type of null columns as Unknown

### LINKED ISSUE #1044: Use Unknown logical type for null columns
Based off discussion in #1038, it makes more sense to infer columns that are entirely null (i.e. made entirely up of `None`, `np.nan`, or `pd.NA`) as `Unknown` logical type instead of `Double` or `Categorical`. This way, entirely null columns are all handled in a consistent manner.

### PR DESCRIPTION
**Pull request by frances-h**
- Columns that are entirely null (made up of `None`, `np.nan`, or `pd.NA`) are inferred as `Unknown` instead of `Double`/`Categorical`.
- Closes #1044

### CODE REVIEW

File: woodwork/type_sys/type_system.py
Comment (tamargrey) [inline woodwork/type_sys/type_system.py:265]: does this catch the column of `'nan'` strings? Or are we going to handle that in a separate PR?

Comment (frances-h) [reply woodwork/type_sys/type_system.py:265]: It only catches true null values, not the string nan representations. That's going to get handled in a separate PR once we figure out how we want to handle them. This does mean that the 'nan' -> Datetime bug is still present. I could add the check here that discards Datetime as a logical type if the entire column is NaT and that would at least avoid the issue for now.

Comment (tamargrey) [reply woodwork/type_sys/type_system.py:265]: cool, just double checking! Don't need to handle it right now, but maybe it's worth this PR having its own issue outlining the feature change to have fully null columns be Unknown

File: woodwork/tests/type_system/test_ltype_inference.py
Comment (tamargrey) [inline woodwork/tests/type_system/test_ltype_inference.py:145]: we should also confirm that `inferred_type.transform(series)` works as expected for all of these series (though maybe that should go in `test_logical_types.py`)

File: woodwork/tests/type_system/test_ltype_inference.py
Comment (tamargrey) [inline woodwork/tests/type_system/test_ltype_inference.py]: I think that the `series.astype(dtype)` will be redundant -- `LogicalType.transform` will perform the astype call if needed

**Review by tamargrey — APPROVED** [review]
lgtm once changes are merged in from main!

**Review by gsheni — APPROVED** [review]

**Pull request merged**
